### PR TITLE
fix bulk action slow query

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

File: packages/api/src/resolvers/article/index.ts [hunk 1]
```diff
@@ -792,6 +791,12 @@ export const bulkActionResolver = authorized<
       },
     })
 
+    // parse query
+    const searchQuery = parseSearchQuery(query)
+    if (searchQuery.ids.length > 100) {
```
Comment (jacksonh) [inline packages/api/src/resolvers/article/index.ts:796]: 👍

File: packages/api/src/services/library_item.ts [hunk 2]
```diff
@@ -133,37 +120,35 @@ const buildWhereClause = (
       case InFilter.SUBSCRIPTION:
         queryBuilder
           .andWhere("NOT ('library' ILIKE ANY (library_item.label_names))")
-          .andWhere({
-            subscription: Not(IsNull()),
-            archivedAt: IsNull(),
-          })
+          .andWhere('library_item.archived_at IS NULL')
+          .andWhere('library_item.subscription IS NOT NULL')
```
Comment (jacksonh) [inline packages/api/src/services/library_item.ts:124]: nice, seems like that ILIKE above was a bug.

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/services/library_item.ts (hunk 2 above)
Comment (sywhb) [reply packages/api/src/services/library_item.ts:124]: sql looks okay. let me test it in the db

Comment (sywhb) [reply packages/api/src/services/library_item.ts:124]: verified in database and it is not a bug

**Pull request merged**